schema/pod.go

```diff
@@ -116,6 +116,11 @@ func (al AppList) assertValid() error {
 			return fmt.Errorf(`duplicate apps of name %q`, a.Name)
 		}
 		seen[a.Name] = true
+		if a.App != nil {
```

Comment:
It looks like we have a test-case exercising exactly this code-path, and panics here.